rdmo/questions/rules.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import rules
+
+from rdmo.projects.rules import is_project_member, is_site_manager
+
+logger = logging.getLogger(__name__)
+
+@rules.predicate
+def can_read_element(user, obj) -> bool:
+    ''' Checks if the user is an editor for the sites to which this element is readable '''
+    
+    if not hasattr(obj, 'sites'):
+        logger.debug('questions.rules.%s: obj %s has no attribute %s', 'can_read_element', obj, 'sites')
+        return False
+
+    return user.role.editor.filter(id__in=obj.sites.all()).exists()
+
+@rules.predicate
+def can_edit_element(user, obj) -> bool:
+    ''' Checks if the user is an editor for the sites to which this element is editable '''
+
+    if not hasattr(obj, 'editors'):
+        logger.debug('questions.rules.%s: obj %s has no attribute %s', 'can_edit_element', obj, 'editors')
+        return False
+    
+    return user.role.editor.filter(id__in=obj.editors.all()).exists()
+
+@rules.predicate
```

Comment:
The last tree roles should be covered by model-based-permissions, see comment for `rdmo/accounts/models.py`.